sshd/net.go

```diff
@@ -32,12 +33,18 @@ func (l *SSHListener) handleConn(conn net.Conn) (*Terminal, error) {
 		conn = ReadLimitConn(conn, l.RateLimit())
 	}
 
+	// Handshake shouldn't take more than 10 seconds
+	conn.SetReadDeadline(time.Now().Add(10 * time.Second))
```

Comment:
Should this be `conn.SetDeadline`?

Also IMO 60 seconds is probably safer.

Reply:
It's still not ideal since a client could drip a byte every N seconds and stay connected indefinitely without joining ssh-chat, but it's certainly an improvement.

Reply:
`SetDeadline` sets both read and write deadline, if write takes a long time then it's probably the connections fault.
The ssh protocol states that the first thing to be done is exchange versions, we write our version and then try to read the client version byte by byte hence the read deadline tells us if the client is not responding.

Yes, technically the client can stay connected but the assumption is that they are bots scanning ports mindlessly and not targeted attacks which know what to write.